Grocery App/AddCategoryViewController.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  AddCategoryViewController.swift
+//  Grocery App
+//
+//  Created by Souryadeep Sadhukhan on 06/05/21.
+//
+
+import UIKit
+import FirebaseAuth
+import FirebaseCore
+import FirebaseFirestore
+protocol PassAction {
+    func addTapped(Name: String)
+}
+class AddCategoryViewController: UIViewController, UITextFieldDelegate {
+
+    @IBOutlet weak var newCategoryTxtField: UITextField!
+    @IBOutlet weak var newRankTxtField: UITextField!
+    var uid = "nil"
+    var category = " "
+    var rank = " "
+    var selectionDelegate: PassAction!
```

Comment:
Delegate property must be optional not forcefully unwrapped.

Reply:
Done Sir